backend/services/implementations/courseService.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable class-methods-use-this */
+import {
+  CourseUnitDTO,
+  CreateCourseUnitDTO,
+  UpdateCourseUnitDTO,
+} from "../../types/courseTypes";
+import logger from "../../utilities/logger";
+import ICourseService from "../interfaces/courseService";
+import MgCourseUnit, { CourseUnit } from "../../models/courseunit.mgmodel";
+import { getErrorMessage } from "../../utilities/errorUtils";
+
+const Logger = logger(__filename);
+
+class CourseService implements ICourseService {
+  async getCourses(): Promise<Array<CourseUnitDTO>> {
+    try {
+      const courses: Array<CourseUnit> = await MgCourseUnit.find();
+      return courses.map((course) => ({
+        id: course.id,
+        displayIndex: course.displayIndex,
+        title: course.title,
+        modules: course.modules,
+      }));
+    } catch (error) {
+      Logger.error(
+        `Failed to get courses, . Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+  }
+
+  async createCourse(course: CreateCourseUnitDTO): Promise<CourseUnitDTO> {
+    let newCourse: CourseUnit | null;
+    try {
+      const largestDisplayIndexCourse = await MgCourseUnit.findOne().sort({
```

Comment:
hey wait - what if we use the number of documents as our largest display index? [countDocument?](https://mongoosejs.com/docs/api/model.html#Model.countDocuments())

we're preserving the invariant that our N units are numbered from 1 to N in our delete method, so I think we can rely on that